lighthouse-cli/bin.ts

```diff
@@ -252,25 +251,42 @@ function handleError(err: LighthouseError) {
 }
 
 function runLighthouse(url: string,
-                       flags: {port: number, skipAutolaunch: boolean, selectChrome: boolean, output: any, outputPath: string, view: boolean},
+                       flags: {port: number, skipAutolaunch: boolean, selectChrome: boolean, output: any,
+                         outputPath: string, view: boolean, saveArtifacts: boolean, saveAssets: boolean},
                        config: Object): Promise<undefined> {
 
   let chromeLauncher: ChromeLauncher;
   return initPort(flags)
     .then(() => getDebuggableChrome(flags))
     .then(chrome => chromeLauncher = chrome)
     .then(() => lighthouse(url, flags, config))
+    .then((results: Results) => {
+      if (flags.saveArtifacts) {
+        assetSaver.saveArtifacts(results.artifacts);
+      }
+      if (flags.saveAssets) {
+        return assetSaver.saveAssets({url: results.url}, results.artifacts).then(() => results);
+      }
+      return results;
+    })
+    .then((results: Results) => {
+      // remove artifacts so users won't get unnecessary artifacts in the report.
```

Comment:
Thanks for your feedback. But moving it to previous `.then()` means we need to delete artifacts separately in 2 places, when `flags.saveAssets === true` and when `flags.saveAssets===false`, because `assetSaver.saveAssets` is async. Do you have any advice to move that line to previous `.then()` while not having it in 2 places?